src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java

```diff
@@ -0,0 +1,148 @@
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import edu.ucsb.cs156.organic.errors.EntityNotFoundException;
+
+import javax.validation.Valid;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+@Tag(name = "Courses")
+@RequestMapping("/api/courses")
+@RestController
+@Slf4j
+public class CoursesController extends ApiController {
+
+    @Autowired
+    CourseRepository courseRepository;
+
+    @Autowired
+    StaffRepository courseStaffRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Operation(summary = "List all courses")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
+    @GetMapping("/all")
+    public Iterable<Course> allCourses() {
+        User u = getCurrentUser().getUser();
+        log.info("u={}", u);
+        if (u.isAdmin()) {
+            return courseRepository.findAll();
+        } else {
+            return courseRepository.findCoursesStaffedByUser(u.getGithubId());
+        }
+    }
+
+    @Operation(summary = "Create a new course")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public Course postCourse(
+            @Parameter(name = "name", description ="course name, e.g. CMPSC 156" ) @RequestParam String name,
+            @Parameter(name = "school", description ="school abbreviation e.g. ucsb" ) @RequestParam String school,
+            @Parameter(name = "term", description = "quarter or semester, e.g. F23") @RequestParam String term,
+            @Parameter(name = "start", description = "in iso format, i.e. YYYY-mm-ddTHH:MM:SS; e.g. 2023-10-01T00:00:00 see https://en.wikipedia.org/wiki/ISO_8601") @RequestParam("start") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime start,
+            @Parameter(name = "end", description = "in iso format, i.e. YYYY-mm-ddTHH:MM:SS; e.g. 2023-12-31T11:59:59 see https://en.wikipedia.org/wiki/ISO_8601") @RequestParam("end") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime end,
+            @Parameter(name = "githubOrg", description = "for example ucsb-cs156-f23" ) @RequestParam String githubOrg)
+            throws JsonProcessingException {
+
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
```

Comment:
Do you need this comment here since it is mentioned in the parameters above

Reply:
Maybe not, but it's not hurting anything much.  I might remove it if I'm in making other changes.

Reply:
removed